src/wags_tails/guide_to_pharmacology.py

```diff
@@ -0,0 +1,112 @@
+"""Provide source fetching for Guide To Pharmacology."""
+import logging
+import re
+from pathlib import Path
+from typing import NamedTuple, Tuple
+
+import requests
+
+from .base_source import DataSource, RemoteDataError
+from .utils.downloads import download_http
+from .utils.storage import get_latest_local_file
+from .utils.versioning import parse_file_version
+
+_logger = logging.getLogger(__name__)
+
+
+class GtoPLigandPaths(NamedTuple):
+    """Container for GuideToPharmacology file paths."""
+
+    ligands: Path
+    ligand_id_mapping: Path
+
+
+class GToPLigandData(DataSource):
+    """Provide access to Guide to Pharmacology data."""
+
+    _src_name = "guidetopharmacology"
+    _filetype = "tsv"
+
+    @staticmethod
+    def _get_latest_version() -> str:
+        """Retrieve latest version value
+
+        :return: latest release value
+        :raise RemoteDataError: if unable to parse version number from releases API
+        """
+        r = requests.get("https://www.guidetopharmacology.org/")
+        r.raise_for_status()
+        r_text = r.text.split("\n")
+        pattern = re.compile(r"Current Release Version (\d{4}\.\d) \(.*\)")
+        for line in r_text:
+            if "Current Release Version" in line:
+                matches = re.findall(pattern, line.strip())
+                if matches:
+                    return matches[0]
+        else:
+            raise RemoteDataError(
+                "Unable to parse latest Guide to Pharmacology version number homepage HTML."
+            )
+
+    def _download_data(self, file_paths: GtoPLigandPaths) -> None:
+        """Perform file downloads.
+
+        :param file_paths: locations to save files at
+        """
+        download_http(
+            "https://www.guidetopharmacology.org/DATA/ligands.tsv",
+            file_paths.ligands,
+            tqdm_params=self._tqdm_params,
+        )
+        download_http(
+            "https://www.guidetopharmacology.org/DATA/ligand_id_mapping.tsv",
+            file_paths.ligand_id_mapping,
+            tqdm_params=self._tqdm_params,
+        )
+
+    def get_latest(
+        self, from_local: bool = False, force_refresh: bool = False
+    ) -> Tuple[GtoPLigandPaths, str]:
```

Comment:
hemonc and gtop both acquire multiple files. With HemOnc, it's basically a bundle (each one is meaningless without the others) but with GtoP, it's just a selection of useful files (and in fact, DGIdb uses a different selection of GtoP files). I thought about a few different ways to handle this (including making a separate class for each file) and all of them were sort of awkward. I think returning a namedtuple of paths like this is the easiest way to handle it for data consumers.